### PR TITLE
fix: Fix spurious author extraction errors

### PR DESCRIPTION
**Pull request by larseggert**
This fixes edge cases including the following names:
* Joris van Kuijc van Malsen
* दिल
* शिला के.सी.
* Lady Garcia
* Carolina de la O
* உயிரோவியம் தங்கராஐ;
* Alexandre Koch II
* இந்திராணி ஆதர்ஷ்,ஆதேஷ்

### CODE REVIEW

**Comment by larseggert** [comment]
@rjsparks I'm still running the tests in a loop to generate more failures, but I think this is ready for review in the meantime.

File: ietf/utils/draft.py [hunk 1]
```diff
@@ -820,8 +825,9 @@ def dotexp(s):
                     first, last = author.rsplit(".", 1)
                     first += "."
                 else:
-                    author = "[A-Z].+ " + author
-                    first, last = author.rsplit(" ", 1)
+                    # handle single-word names
```
Comment (jennifer-richards) [inline ietf/utils/draft.py:828]: Was the `[A-Z].+` doing anything functional before?

**Review by jennifer-richards — APPROVED** [review]
One question, mostly out of curiosity, but this looks ok to me.

**Comment by larseggert** [comment]
Found another case, where "இந்திராணி ஆதர்ஷ்,ஆதேஷ்" renders as "Intiraanni Aatrss,aateess" and the comma is confusing the current regex.

File: ietf/utils/draft.py (hunk 1 above)
Comment (larseggert) [reply ietf/utils/draft.py:828]: I don't think so. If we end up in that part of the conditional, `author` does not contain a space, so it is a single word. So we'd never match a first name.

**Comment by larseggert** [comment]
No more failures in the last few hours.

File: ietf/utils/draft.py (hunk 1 above)
Comment (rjsparks) [reply ietf/utils/draft.py:828]: It looks like a bug to me. At the very least, the old code was a strange construct and I suspect it is an edit-fail from long ago.

Comment (jennifer-richards) [reply ietf/utils/draft.py:828]: Thanks, that's what it looked like but the code is a bit tricky (to say the least).

**Review by rjsparks — APPROVED** [review]

**Pull request merged**